runtime/src/main/java/org/corfudb/runtime/view/stream/StreamAddressSpace.java

```diff
@@ -84,34 +79,30 @@ public boolean contains(long address) {
      * @return returns a as the merged StreamAddressSpace
      */
     public static StreamAddressSpace merge(StreamAddressSpace a, StreamAddressSpace b) {
-        a.mergeFrom(b);
-        return a;
-    }
-
-    // Merge another StreamAddressSpace into this instance
-    private void mergeFrom(StreamAddressSpace other) {
-        this.trimMark = Long.max(trimMark, other.getTrimMark());
-        this.bitmap.or(other.bitmap);
+        a.trimMark = Long.max(a.trimMark, b.getTrimMark());
+        a.bitmap.or(b.bitmap);
 
         // Because the trim mark can increase after the merge another
         // trim needs to be issued
-        this.trim(this.trimMark);
+        a.trim(a.trimMark);
```

Comment:
It is. `a.trimMark = Long.max(a.trimMark, b.getTrimMark());` , but I think its better to set the new trim mark through by calling trim instead of setting the field directly.

Reply:
Sometimes we need to force the trim mark, because the trim method doesn't update the trimMark if there are no addresses to trim. I added a comment.

Reply:
Added a test `testMergeDifferentTrimMarks`